### PR TITLE
refactor: UI options

### PR DESCRIPTION
**Pull request by MarcGonzalezMoratona**
Improved readability in UI Options

### CODE REVIEW

File: Game/Scripts/UIOptionsMenu.cpp [hunk 1]
```diff
@@ -466,66 +466,64 @@ void UIOptionsMenu::IsSizeOptionEnabled()
 {
 	float4 colorSet;
 
-	int gameOptionWindowsMode = actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[static_cast<int>(Button::WINDOWSMODE)].actualOption;
+	int gameCanvasIndex = static_cast<int>(Canvas::GAME_CANVAS);
+	int resolutionIndex = static_cast<int>(Button::RESOLUTION);
+	int gameOptionWindowsMode = actualConfig[gameCanvasIndex].options[static_cast<int>(Button::WINDOWSMODE)].actualOption;
 
+	GameObject* resolutionParentGameObject = buttonsAndCanvas[gameCanvasIndex].canvas->
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UIOptionsMenu.cpp:473]: is resolutionParentGameObject a descriptive name related to the functionality? I wanted to store the Gameobject in a variable but I'm not 100% of what this GameObject represents

File: Game/Scripts/UIOptionsMenu.cpp [hunk 2]
```diff
@@ -466,66 +466,64 @@ void UIOptionsMenu::IsSizeOptionEnabled()
...
 }
 
 void UIOptionsMenu::IsFpsEnabled()
 {
 	float4 colorSet;
-	if (actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[(int)Button::VSYNC].actualOption == 1)
+	int gameCanvasIndex = static_cast<int>(Canvas::GAME_CANVAS);
+	int FPSIndex = static_cast<int>(Button::FPS);
+
+	GameObject* FPSParentGameObject = buttonsAndCanvas[gameCanvasIndex].canvas->GetChildren()[FPSIndex]->GetChildren()[1];
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UIOptionsMenu.cpp:501]: is FPSParentGameObject a descriptive name related to the functionality? I wanted to store the Gameobject in a variable but I'm not 100% of what this GameObject represents

File: Game/Scripts/UIOptionsMenu.cpp (hunk 1 above)
Comment (Piterga12) [reply Game/Scripts/UIOptionsMenu.cpp:473]: Yeah, i think it should be fine, is a tricky one, can't think of a better name, so should be fine

**Review by Piterga12 — APPROVED** [review]

**Review by Piterga12 — DISMISSED** [review]

File: Game/Scripts/UIOptionsMenu.cpp [hunk 3]
```diff
@@ -466,67 +463,64 @@ void UIOptionsMenu::IsSizeOptionEnabled()
 {
 	float4 colorSet;
 
-	int gameOptionWindowsMode = actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[static_cast<int>(Button::WINDOWSMODE)].actualOption;
+	int gameCanvasIndex = static_cast<int>(Canvas::GAME_CANVAS);
+	int resolutionIndex = static_cast<int>(Button::RESOLUTION);
+	int gameOptionWindowsMode = actualConfig[gameCanvasIndex].options[static_cast<int>(Button::WINDOWSMODE)].actualOption;
```
Comment (Vlukkee) [inline Game/Scripts/UIOptionsMenu.cpp:468]: If you want change this delete the static cast and add the number, its not worth it have the two names.

Example:
``
int gameCanvasIndex = 0;
int resolutionIndex = 3;
int windowModeIndex = 4;
int gameOptionWindowsMode =actualConfig[gameCanvasIndex].options[windowModeIndex].actualOption;``

**Comment by Vlukkee** [comment]
Thanks for making the code more readable! If my explanatios are not good please let me know and i try to explain better.

File: Game/Scripts/UIOptionsMenu.cpp (hunk 3 above)
Comment (MarcGonzalezMoratona) [reply Game/Scripts/UIOptionsMenu.cpp]: done!

**Review by AuxPort66 — APPROVED** [review]

**Review by Vlukkee — APPROVED** [review]

File: Game/Scripts/UIOptionsMenu.cpp [hunk 4]
```diff
@@ -466,67 +463,65 @@ void UIOptionsMenu::IsSizeOptionEnabled()
...
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->
-			GetChildren()[static_cast<int>(Button::RESOLUTION)]->GetChildren()[1]->GetChildren()[2]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->
-			GetChildren()[static_cast<int>(Button::RESOLUTION)]->GetChildren()[1]->GetChildren()[3]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->
-			GetChildren()[static_cast<int>(Button::RESOLUTION)]->GetChildren()[1]->GetChildren()[4]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->
-			GetChildren()[static_cast<int>(Button::RESOLUTION)]->GetChildren()[1]->GetChildren()[0]->Enable();
-		actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[static_cast<int>(Button::RESOLUTION)].locked = true;
+		resolutionParentGameObject->GetChildren()[1]->Disable();
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:477]: We can probably use a `fori` starting from index 1.

File: Game/Scripts/UIOptionsMenu.cpp [hunk 5]
```diff
@@ -466,67 +463,65 @@ void UIOptionsMenu::IsSizeOptionEnabled()
...
-		GetChildren()[1]->GetChildren()[0]->GetComponent<ComponentImage>()->SetColor(colorSet);
+	resolutionParentGameObject->GetChildren()[0]->GetComponent<ComponentImage>()->SetColor(colorSet);
 }
 
 void UIOptionsMenu::IsFpsEnabled()
 {
 	float4 colorSet;
-	if (actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[(int)Button::VSYNC].actualOption == 1)
+	int gameCanvasIndex = 0;
+	int FPSIndex = 0;
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:497]: Same thing as before. Centralize in a namespace and make them constexpressions, will make it easier imo

File: Game/Scripts/UIOptionsMenu.cpp [hunk 6]
```diff
@@ -466,67 +463,65 @@ void UIOptionsMenu::IsSizeOptionEnabled()
...
+
+	GameObject* FPSParentGameObject = buttonsAndCanvas[gameCanvasIndex].canvas->GetChildren()[FPSIndex]->GetChildren()[1];
+	
+	if (actualConfig[gameCanvasIndex].options[(int)Button::VSYNC].actualOption == 1)
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:501]: Why do you use the enum here?

File: Game/Scripts/UIOptionsMenu.cpp [hunk 7]
```diff
@@ -466,67 +463,65 @@ void UIOptionsMenu::IsSizeOptionEnabled()
...
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->GetChildren()[static_cast<int>(Button::FPS)]->
-			GetChildren()[1]->GetChildren()[1]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->GetChildren()[static_cast<int>(Button::FPS)]->
-			GetChildren()[1]->GetChildren()[2]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->GetChildren()[static_cast<int>(Button::FPS)]->
-			GetChildren()[1]->GetChildren()[3]->Disable();
-		buttonsAndCanvas[static_cast<int>(Canvas::GAME_CANVAS)].canvas->GetChildren()[static_cast<int>(Button::FPS)]->
-			GetChildren()[1]->GetChildren()[0]->Enable();
-		actualConfig[static_cast<int>(Canvas::GAME_CANVAS)].options[static_cast<int>(Button::FPS)].locked = true;
+		FPSParentGameObject->GetChildren()[1]->Disable();
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:504]: Same, just loop.

File: Game/Scripts/UIOptionsMenu.cpp [hunk 8]
```diff
@@ -563,9 +559,10 @@ void UIOptionsMenu::GameOption(int button, int option)
 		default:
 			break;
 		}
-			IsFpsEnabled();
+		IsFpsEnabled();
 		break;
-	case 2: //BRIGHTNESS
+
+	case 2: // BRIGHTNESS
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:565]: Use the enum for the cases (and the parameter). As a rule of thumb, don't write comments that can be explained by the code itself.

File: Game/Scripts/UIOptionsMenu.cpp [hunk 9]
```diff
@@ -666,21 +666,21 @@ void UIOptionsMenu::AudioOption(int button, int option)
 {
 	switch (button)
 	{
-	case 0: //MASTER
+	case 0: // MASTER
```
Comment (gerard-queralt) [inline Game/Scripts/UIOptionsMenu.cpp:669]: Same here.

**Review by Vlukkee — APPROVED** [review]

**Review by Piterga12 — APPROVED** [review]

**Pull request merged**